### PR TITLE
Style cards changes

### PR DESCRIPTION
**Pull request by jansule**
## Description

This adds a new card-based layout that does not make use of any windows. Multiple components have been added.

This PR was done in collaboration with @annarieger.

Please note that there are still a few issues regarding state handling. Drag and drop and some polishing is missing as well. This will be provided in a follow up PR.

geostyler-card-layout

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [BSD 2-Clause License](https://github.com/geostyler/geostyler/blob/master/)
- [x] I have followed the [guidelines for contributing](https://github.com/geostyler/geostyler/blob/master/CONTRIBUTING.md)
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/geostyler/geostyler/blob/master/CODE_OF_CONDUCT.md)
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: src/Component/CardStyle/CardStyle.tsx [hunk 1]
```diff
@@ -82,131 +76,318 @@ interface CardStyleDefaultProps {
...
+  const styleOverviewPropsOverwrites = {
+    rulesProps: {
+      ruleCardProps: {
+        rendererProps: {
+          rendererType: rendererType
+        }
+      }
+    }
+  };
+  const mergedStyleOverviewProps = _merge(styleOverviewProps, styleOverviewPropsOverwrites);
```
Comment (KaiVolland) [inline src/Component/CardStyle/CardStyle.tsx:302]: No lodas needed herer i guess.
```suggestion
  const mergedStyleOverviewProps = {...styleOverviewProps, ...styleOverviewPropsOverwrites};
```

File: src/Component/Renderer/Renderer.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+export const Renderer: React.FC<RendererProps> = ({
+  rendererType = 'OpenLayers',
+  onSymbolizerClick = () => {},
+  data,
+  sldRendererProps,
+  symbolizers
+}) => {
+
+  return (
```
Comment (KaiVolland) [inline src/Component/Renderer/Renderer.tsx:65]: You could move the check/switch `rendererType === 'OpenLayers'` before the `return` to enhance readability.

File: src/Component/RuleOverview/RuleOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    rendererProps: {
+      symbolizers: rule.symbolizers,
+      data: data
+    }
+  };
+  const mergedRuleFieldContainerProps = _merge(ruleFieldContainerProps, ruleFieldContainerPropsOverwrites);
+
+  const symbolizersPropsOverwrites = {
+  };
+  const mergedSymbolizersProps = _merge(symbolizersProps, symbolizersPropsOverwrites);
```
Comment (KaiVolland) [inline src/Component/RuleOverview/RuleOverview.tsx:138]: Can be replaced with spread operator syntax.

File: src/Component/Rules/Rules.tsx [hunk 4]
```diff
@@ -218,24 +218,32 @@ export const Rules: React.FC<RulesProps> = ({
...
-      {locale.addRule}
-    </Button>,
-    <Button
-      onClick={classificationClick}
-      key={1}
-    >
-      {locale.classification}
-    </Button>
-  ];
+  const getDefaultActions = (): ReactNode[] => {
```
Comment (KaiVolland) [inline src/Component/Rules/Rules.tsx:226]: I prefer the approach of just defining a variable instead of a function if no arguments are provided.

**Review by KaiVolland — APPROVED** [review]
Nice job. Just some minor comments.

File: src/Component/RuleOverview/RuleOverview.tsx (hunk 3 above)
Comment (jansule) [reply src/Component/RuleOverview/RuleOverview.tsx]: solved in https://github.com/geostyler/geostyler/pull/1507/commits/3e057aa39664e4e89fcdd875c32c996f8ed757fd

**Pull request merged**